src/utils.cpp

```diff
@@ -360,4 +360,257 @@ int mega_snprintf(char *s, size_t n, const char *format, ...)
 }
 #endif
 
+string TLVstore::TLVrecordsToContainer(SymmCipher *key, encryptionmode_t mode)
+{
+    // serialize the TLV records
+    string container = TLVrecordsToContainer();
+
+    // encrypt the result
+    int ivlen, taglen;
+    bool useCCM = false;
+    switch (mode)
+    {
+    case AES_CCM_12_16:
+    case AES_GCM_12_16_BROKEN:
+        ivlen = 12;
+        taglen = 16;
+        useCCM = true;
+        break;
+
+    case AES_CCM_10_16:
+        ivlen = 10;
+        taglen = 16;
+        useCCM = true;
```

Comment:
this seems quite "binary". What if we were to use not AES-CCM and AES-GCM at some point, but also offer Salsa20 with Poly1305? But then I guess we can refactor ... :-)